### PR TITLE
[FIX] cooperator: Do not search account 416000

### LINKED ISSUE #251: Refactor `get_accounting_account` :  do not look for 416000
In https://github.com/coopiteasy/vertical-cooperative/blob/1875bed073c6c49efef12f4de642eaef94ea5ec3/easy_my_coop/models/subscription_request.py#L534 : do not rely on 416000 as it is specific to Belgian charts of account. Rely solely on configuration instead.

### PR DESCRIPTION
**Pull request by carmenbianca**
## Description

Fixes #251 

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: cooperator/models/subscription_request.py [hunk 1]
```diff
@@ -531,16 +531,11 @@ def get_journal(self):
...
+        if account:
+            return account
         else:
-            accounts = account_obj.search([("code", "=", "416000")])
-            if accounts:
-                account = accounts[0]
-            else:
-                raise UserError(_("You must set a cooperator account on you company."))
-        return account
+            raise UserError(_("You must set a cooperator account on you company."))
```
Comment (robinkeunen) [inline cooperator/models/subscription_request.py]: typo: you**R** company

File: l10n_be_cooperator/models/subscription_request.py [hunk 2]
```diff
@@ -8,6 +9,16 @@ class SubscriptionRequest(models.Model):
...
+    def get_accounting_account(self):
+        try:
+            return super().get_accounting_account()
+        except UserError as error:
+            accounts = self.env["account.account"].search([("code", "=", "416000")])
+            if accounts:
+                return accounts[0]
+            else:
+                raise error
+
```
Comment (robinkeunen) [inline l10n_be_cooperator/models/subscription_request.py]: I would make things simpler : remove this and solely rely on module configuration.

File: cooperator/models/subscription_request.py [hunk 3]
```diff
@@ -531,16 +531,11 @@ def get_journal(self):
         return self.env.ref("cooperator.subscription_journal")
 
     def get_accounting_account(self):
-        account_obj = self.env["account.account"]
-        if self.company_id.property_cooperator_account:
-            account = self.company_id.property_cooperator_account
+        account = self.company_id.property_cooperator_account
```
Comment (robinkeunen) [inline cooperator/models/subscription_request.py:535]: Can you add a note in the readme to tell the user to configure this account ?

**Review by robinkeunen — CHANGES_REQUESTED** [review]
Left some comments.

@vdewulf 
**before this PR**
`cooperator` fetches the cooperator account
1. on the `property_cooperator_account` configuration on the company
2. if not found fetches 416000
3. if not found raises an error

**After this PR**
`cooperator` fetches the cooperator account
1. on the `property_cooperator_account` configuration field on the company
2. if not found AND l10n_be_cooperator is installed, fetches 416000
3. if not found raises an error

My question : can we make things simpler and solely rely on the configuration ? And therefore remove the code fetching account 416000 in l10n_be_cooperator ?

**Comment by robinkeunen** [comment]
> Left some comments.
> 
> @vdewulf **before this PR** `cooperator` fetches the cooperator account
> 
> 1. on the `property_cooperator_account` configuration on the company
> 2. if not found fetches 416000
> 3. if not found raises an error
> 
> **After this PR** `cooperator` fetches the cooperator account
> 
> 1. on the `property_cooperator_account` configuration field on the company
> 2. if not found AND l10n_be_cooperator is installed, fetches 416000
> 3. if not found raises an error
> 
> My question : can we make things simpler and solely rely on the configuration ? And therefore remove the code fetching account 416000 in l10n_be_cooperator ?

@vdewulf Did you have a chance to look into this ?

**Comment by vdewulf** [comment]
@robinkeunen sorry, I tought I asnwered that.
As this field (account cooperator) is mandatory in the company config (correct me if I'm wrong), I think it's sufficient to just fetch it there, indeed.

**Review by robinkeunen — APPROVED** [review]

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**